src/magic_ter_fur_transform.cpp

```diff
@@ -0,0 +1,287 @@
+                               weighted_int_list<furn_id> &list )
+{
+    if( jsi.has_string( json_key ) ) {
+        list.add( furn_id( jsi.get_string( json_key ) ), 1 );
+        return;
+    }
+    JsonArray jarr = jsi.get_array( json_key );
+    while( jarr.has_more() ) {
+        if( jarr.test_array() ) {
+            JsonArray inner = jarr.next_array();
+            list.add( furn_id( inner.get_string( 0 ) ), inner.get_int( 1 ) );
+        } else {
+            list.add( furn_id( jarr.next_string() ), 1 );
+        }
+    }
+}
+
+void ter_furn_transform::load( JsonObject &jo, const std::string & )
+{
+    std::string input;
+    mandatory( jo, was_loaded, "id", input );
+    id = ter_furn_transform_id( input );
+    optional( jo, was_loaded, "fail_message", fail_message, "" );
+
+    if( jo.has_member( "terrain" ) ) {
+        JsonArray obj_array = jo.get_array( "terrain" );
+        while( obj_array.has_more() ) {
+            JsonObject ter_obj = obj_array.next_object();
+            JsonArray target_array = ter_obj.get_array( "valid_terrain" );
+
+            weighted_int_list<ter_id> results;
+            load_ter_results( ter_obj, "result", results );
+            const std::string message = ter_obj.get_string( "message" );
+            bool message_good = true;
+            if( ter_obj.has_bool( "message_good" ) ) {
+                message_good = ter_obj.get_bool( "message_good" );
+            }
+
+            while( target_array.has_more() ) {
+                const std::string valid_terrain = target_array.next_string();
+                ter_transform.emplace( ter_id( valid_terrain ), results );
+                if( !message.empty() ) {
+                    ter_message.emplace( ter_id( valid_terrain ), std::make_pair( message, message_good ) );
+                }
+            }
+
+            target_array = ter_obj.get_array( "valid_flags" );
+
+            while( target_array.has_more() ) {
+                const std::string valid_terrain = target_array.next_string();
+                ter_flag_transform.emplace( valid_terrain, results );
+                if( !message.empty() ) {
+                    ter_flag_message.emplace( valid_terrain, std::make_pair( message, message_good ) );
+                }
+            }
+        }
+    }
+
+    if( jo.has_member( "furniture" ) ) {
+        JsonArray obj_array = jo.get_array( "furniture" );
+        while( obj_array.has_more() ) {
+            JsonObject furn_obj = obj_array.next_object();
+            JsonArray target_array = furn_obj.get_array( "valid_terrain" );
+
+            weighted_int_list<furn_id> results;
+            load_furn_results( furn_obj, "result", results );
+            const std::string message = furn_obj.get_string( "message" );
+            bool message_good = true;
+            if( furn_obj.has_bool( "message_good" ) ) {
+                message_good = furn_obj.get_bool( "message_good" );
```

Comment:
```suggestion
            const bool message_good = furn_obj.get_bool( "message_good", true );
```
Instead of the whole `if` clause.